clap-utils/src/keypair.rs

```diff
@@ -0,0 +1,78 @@
+use bip39::{Language, Mnemonic, Seed};
+use clap::values_t;
+use rpassword::prompt_password_stderr;
+use solana_sdk::signature::{
+    keypair_from_mnemonic_and_passphrase, keypair_from_seed, read_keypair_file, Keypair,
+    KeypairUtil,
+};
+use std::{
+    error,
+    io::{stdin, stdout, Write},
+};
+
+/// Reads user input from stdin to retrieve a mnemonic and passphrase for keypair derivation
+pub fn keypair_from_mnemonic(keypair_name: &str) -> Result<Keypair, Box<dyn error::Error>> {
+    let mnemonic_phrase =
+        prompt_password_stderr(&format!("[{}] mnemonic phrase: ", keypair_name)).unwrap();
+    match Mnemonic::from_phrase(mnemonic_phrase.trim(), Language::English) {
+        Ok(mnemonic) => {
+            let passphrase =
+                prompt_password_stderr(&format!("[{}] (optional) passphrase: ", keypair_name))
+                    .unwrap();
+            let seed = Seed::new(&mnemonic, &passphrase);
+            keypair_from_seed(seed.as_bytes())
+        }
+        Err(err) => {
+            print!(
+                "[{}] mnemonic phrase validation failed, continue anyways? (y/n): ",
+                keypair_name
+            );
+            stdout().flush().unwrap();
+
+            let mut buffer = String::new();
+            stdin().read_line(&mut buffer).unwrap();
+            if buffer.trim() != "y" {
+                return Err(err.into());
+            }
+
+            let passphrase =
+                prompt_password_stderr(&format!("[{}] (optional) passphrase: ", keypair_name))
+                    .unwrap();
+
+            keypair_from_mnemonic_and_passphrase(&mnemonic_phrase, &passphrase)
+        }
+    }
+}
```

Comment:
so much `.unwrap()` in here, why not just punt to the caller in all cases with `?`